hls4ml/backends/quartus/passes/transform_types.py

```diff
@@ -27,7 +27,7 @@ def transform(self, model, node):
                 elif node.name in node.model.outputs:
                     new_var = self.struct_var_converter.convert(var, pragma='hls_register', struct_name='outputs')
                 else:
-                    new_var = self.array_var_converter.convert(var, pragma='hls_register')
+                    new_var = self.array_var_converter.convert(var, pragma=' ')
```

Comment:
Should avoid removing this pragma as it affects other layers in Quartus as well. 

On the other hand, having this pragma causes HLS compilation to fail with a message along the lines of "Failed to place variable in registers, you might be taking a reference or a pointer to a variable". I've seen this error before, when implementing streams, and I think (this should be easy to verify) a likely cause here is the fact that the LSTM and SimpleRNN methods take arguments as pointers, rather than static C++ arrays of fixed size (even though these two are usually interchangeable) which is how other layers implemented. So, should try changing all method argument pointers to statics arrays.

If this doesn't work, we should further debug this, as it likely decreases performance in other layers

Reply:
Solved!